### ISSUE TITLE
.gitignore located at subdirectories should be respected

### ISSUE THREAD

**Issue by floydwch** [patbenatar/vscode-advanced-new-file#83]
In a micro-service oriented project, it would have different .gitignore files in different subdirectories.

The current implementation of advanced new file collects the .gitignore file from project dir to the system's root, and it doesn't consider the .gitignore files located at subdirectories.

**Comment by patbenatar** [comment]
@floydwch good idea! Perhaps a workaround for the time being: if you open each git repo as its own root in a multi-root workspace, all the gitignores should be honored.

**Comment by floydwch** [comment]
@patbenatar yeah, it's how I'm dealing with it in the meanwhile.
